src/calling/haplotype_abundances.rs

```diff
@@ -3,22 +3,47 @@ use derive_builder::Builder;
 use hdf5;
 use kernel_density;
 use rust_htslib::bcf;
+use serde_json::json;
 
 #[derive(Builder)]
 #[builder(pattern = "owned")]
 pub(crate) struct Caller {
     hdf5_reader: hdf5::File,
     vcf_reader: bcf::Reader,
+    min_tpm: u64,
+    qc_plot: Option<PathBuf>
 }
 
 pub(crate) struct ECDF {
     seqname: String,
     bootstraps: Vec<u64>,
-    ecdf: kernel_density::ecdf::Ecdf<u64>,
+    ecdf: Vec<f64>,
 }
 
-impl Caller {
-    pub(crate) fn cdf(&self, seqname: String) -> Result<ECDF> {
+impl Caller {  
+    //Below function outputs a vector of filtered seqnames according to min_tpm.
+    pub(crate) fn filter_seqnames(&self) -> Result<Vec<String>> {
+        let hdf5 = &self.hdf5_reader;
+        let tpm = hdf5.dataset("bootstrap/bs0")?.read_1d::<u64>()?;
+        let mut indices = Vec::new();
+        for (i,num) in tpm.iter().enumerate() {
+            if num > &self.min_tpm {
+                indices.push(i);
```

Comment:
I think the filtering should not happen based on the bootstrap TPMs, but rather based on the final estimate of kallisto (i.e. the maximum likelihood TPM).

Reply:
Ok, now we use the est_counts. This is already nice. However, they are not normalized for length. Is there a single TPM estimate we could use?